js/src/ast.ts

```diff
@@ -0,0 +1,110 @@
+export type Root = Node[]
+
+export type Node = Collection
```

Comment:
A root node can also be a function, adding another type would also force us to use filter

Reply:
Looking at the Rust we don't actually have a function type a the root node?

```
#[derive(Debug, Serialize, Deserialize, PartialEq)]
#[serde(tag = "kind")]
pub enum RootNode<'a> {
    #[serde(borrow, rename = "collection")]
    Collection(Collection<'a>),
    #[serde(other)]
    Unknown,
}
```

Reply:
That's true, we error if we find one in the parser AST. We can then add `Node = Collection | { kind: unknown }` to force us to filter (so we won't error when a new RootNode variant is added)